packages/pwa/app/utils/url.test.js

```diff
@@ -94,6 +97,71 @@ describe('url builder test', () => {
         const homeUrlDefaultLocale = homeUrlBuilder('/', 'en-GB')
         expect(homeUrlDefaultLocale).toEqual(`/`)
     })
+    test('getUrlWithLocale returns expected for PLP', () => {
+        const location = new URL('http://localhost:3000/it-IT/category/newarrivals-womens')
+
+        delete window.location
+        window.location = location
+
+        const relativeUrl = getUrlWithLocale('fr-FR')
+        expect(relativeUrl).toEqual(`/fr-FR/category/newarrivals-womens`)
+    })
+
+    test('getUrlWithLocale returns expected for PLP without refine param', () => {
+        const location = new URL(
+            'http://localhost:3000/it-IT/category/newarrivals-womens?limit=25&refine=c_refinementColor%3DBianco&sort=best-matches&offset=25'
+        )
+
+        delete window.location
+        window.location = location
+
+        const relativeUrl = getUrlWithLocale('fr-FR', {
+            disallowParams: ['refine']
+        })
+        expect(relativeUrl).toEqual(
+            `/fr-FR/category/newarrivals-womens?limit=25&sort=best-matches&offset=25`
+        )
+    })
+
+    test('getUrlWithLocale returns expected for PLP', () => {
+        const location = new URL('http://localhost:3000/it-IT/category/newarrivals-womens')
+
+        delete window.location
+        window.location = location
+
+        const relativeUrl = getUrlWithLocale('fr-FR')
+        expect(relativeUrl).toEqual(`/fr-FR/category/newarrivals-womens`)
+    })
+
+    test('getUrlWithLocale returns expected for Homepage', () => {
+        const location = new URL('http://localhost:3000/it-IT/')
+
+        delete window.location
+        window.location = location
+
+        const relativeUrl = getUrlWithLocale('fr-FR')
+        expect(relativeUrl).toEqual(`/fr-FR`)
+    })
+
+    test('getUrlWithLocale returns expected for Homepage with Default locale', () => {
+        const location = new URL('http://localhost:3000/it-IT/')
+
+        delete window.location
+        window.location = location
+
+        const relativeUrl = getUrlWithLocale(DEFAULT_LOCALE)
+        expect(relativeUrl).toEqual(`/`)
+    })
+
+    test('getUrlWithLocale returns expected for Homepage without trailing slash', () => {
+        const location = new URL('http://localhost:3000/it-IT')
+
+        delete window.location
+        window.location = location
+
+        const relativeUrl = getUrlWithLocale(DEFAULT_LOCALE)
+        expect(relativeUrl).toEqual(`/`)
+    })
 })
 
 describe('rebuildPathWithParams test', () => {
```

Comment:
I believe at this point, the value for `window.location` is _not_ going to be the original one. Perhaps we can use `beforeEach`/`afterEach` to make sure that location is reset properly.

Reply:
good catch! a typical mistake that is causes effects on another test. I'm using now `beforeEach`/`afterEach`.